addons/mail/models/ir_attachment.py

```diff
@@ -4,33 +4,34 @@
 
 from odoo import _, models, SUPERUSER_ID
 from odoo.exceptions import AccessError, MissingError, UserError
-from odoo.tools import consteq
+from odoo.tools.misc import limited_field_access_token, verify_limited_field_access_token
 from odoo.addons.mail.tools.discuss import Store
 
 
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
-    def _check_attachments_access(self, attachment_tokens):
+    def _has_attachments_ownership(self, attachment_tokens):
         """This method relies on access rules/rights and therefore it should not be called from a sudo env."""
         self = self.sudo(False)
         attachment_tokens = attachment_tokens or ([None] * len(self))
         if len(attachment_tokens) != len(self):
             raise UserError(_("An access token must be provided for each attachment."))
-        for attachment, access_token in zip(self, attachment_tokens):
+        has_ownership = False
+
+        def _is_attachment_owned(attachment, token):
```

Comment:
no need for a private name for an internal method like that

could also be named just `is_owned` if you want a shorter name